source/Core/BSP/Pinecilv2/Setup.cpp

```diff
@@ -80,26 +80,47 @@ void setup_adc(void) {
   adc_cfg.vref           = ADC_VREF_3P2V;
   adc_cfg.resWidth       = ADC_DATA_WIDTH_14_WITH_64_AVERAGE;
```

Comment:
Any chance; you could maybe try and add a block comment before these with a TL;DR of your findings and ∴ why these were selected.

For the future reader :)

Reply:
> because the datasheet says max input to the ADC is 3.2V
I do love this addition to the datasheet not being in the first one I had 🙃 It turned up a while later.

>getMaxTipInC should still be based on the 3.2V limit, because the datasheet says max input to the ADC is 3.2V. 
I agree

>So I think ADC_MAX_RAW_VALUE needs to become a function, not a macro, cause it depends on the "coe" value.

Yep I think so too; It will need to be pulled out to a function and can be a const for other devices.